signing/secp256k1/singlesig/secp256k1.go

```diff
@@ -0,0 +1,69 @@
+package singlesig
+
+import (
+	"crypto/sha256"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/btcsuite/btcd/btcec"
+)
+
+// Secp256k1Signer exposes the signing and verification for ecdsa signature scheme
+type Secp256k1Signer struct {
+}
+
+// Sign is used to sign a message
+func (s *Secp256k1Signer) Sign(private crypto.PrivateKey, msg []byte) ([]byte, error) {
+	if check.IfNil(private) {
+		return nil, crypto.ErrNilPrivateKey
+	}
+
+	privKey, ok := private.Scalar().GetUnderlyingObj().(*btcec.PrivateKey)
+	if !ok {
+		return nil, crypto.ErrInvalidPrivateKey
+	}
+	if len(privKey.Serialize()) != btcec.PrivKeyBytesLen {
+		return nil, crypto.ErrInvalidPrivateKey
+	}
+
+	hash := sha256.Sum256(msg)
+	sig, err := privKey.Sign(hash[:])
+	if err != nil {
+		return nil, err
+	}
+
+	return sig.Serialize(), nil
+}
+
+// Verify is used to verify a signed message
+func (s *Secp256k1Signer) Verify(public crypto.PublicKey, msg []byte, sig []byte) error {
+	if check.IfNil(public) {
+		return crypto.ErrNilPublicKey
+	}
+
+	pubKey, ok := public.Point().GetUnderlyingObj().(*btcec.PublicKey)
+	if !ok {
+		return crypto.ErrInvalidPublicKey
+	}
+	if len(pubKey.SerializeCompressed()) != btcec.PubKeyBytesLenCompressed {
+		return crypto.ErrInvalidPublicKey
+	}
+
+	signature, err := btcec.ParseDERSignature(sig, btcec.S256())
+	if err != nil {
+		return err
+	}
+
+	hash := sha256.Sum256(msg)
```

Comment:
please remove hashing of the message, we need to verify the signature over the exact message.

Reply:
done